invoke-ai/InvokeAI
# Check cuDNN version compatibility on startup

## Summary

This PR adds a cuDNN version check on app startup. If a compatibility issue is detected, a warning message is logged.

Prior to this check, a cuDNN version mismatch would cause the app to silently run with severely-degraded performance.

## QA Instructions

I triggered the error condition by installing two versions of cuDNN in my python environment:
```bash
$ pip freeze | grep cudnn
nvidia-cudnn-cu11==8.5.0.96
nvidia-cudnn-cu12==8.9.2.26
```

On startup, the app now logs the following:
```bash
[2024-03-25 18:58:35,863]::[InvokeAI]::WARNING --> Encountered a cuDNN version issue. This may result in degraded performance. This issue is usually caused by an incompatible cuDNN version installed in your python environment, or on the host system. Full error message:
cuDNN version incompatibility: PyTorch was compiled  against (8, 9, 2) but found runtime version (8, 5, 0). PyTorch already comes bundled with cuDNN. One option to resolving this error is to ensure PyTorch can find the bundled cuDNN. Looks like your LD_LIBRARY_PATH contains incompatible version of cudnn. Please either remove it from the path or install cudnn (8, 9, 2)
```

While running in this degraded state, the app hits a speed of **~4.7 it/s** on the denoising steps with SDXL at 1024x1024.

To fix the issue, I uninstalled `nvidia-cudnn-cu11`. After fixing the underlying issue, the app hits a speed of **~7 it/s** on the denoising steps with SDXL at 1024x1024.


## Merge Plan



## Checklist
